core/src/main/protobuf/sonarlint.proto

```diff
@@ -67,6 +67,7 @@ message Rules {
     bool is_template = 8;
     string template_key = 9;
     string html_note = 10;
+    string type = 11;
```

Comment:
Should we increase storage format version to force people to update their binding? If we do, it may be annoying while it is not really a breaking change. But if we don't, people may wonder why they don't see rule types in their IDE...
IMO: don't update storage version, but remember to precise in release notes that this feature will only work after updating the binding.